### ISSUE TITLE
npm is (erroneously) run when running a uber-jar in production mode.

### ISSUE THREAD

**Issue by jflamy** [vaadin/flow#7111]
- Description of the bug
When running a uber-jar that combines all project dependencies in a single jar, npm should not be run, since everything is present in the jar.  
This is the same bug as #6679 and #6685 which should not have been closed. 

- Root cause
The method `com.vaadin.flow.server.DeploymentConfigurationFactory.getPossibleJarResource(List<URL>)` expects that `webpack.generated.js` will be present >= 2 times in the jars.  This is false when all dependencies have been combined in a single jar using (for example) maven-shade-plugin.

- Minimal reproducible example and  Actual behavior
See https://www.dropbox.com/s/b7urn41vgtz7mrx/owlcms_4.2.9.jar?dl=0 for a uber-jar created with maven-shade-plugin.   
Running this file with `java -jar owlcms_4.2.9.jar` starts npm, which is not correct -- this is meant to be distributed stand-alone or run on a cloud like Heroku.

- Expected behavior
Should start correctly without running npm.
Indeed, the file https://www.dropbox.com/s/f1xdi9b4umnyr66/owlcms_4.2.9_withWorkaround.jar?dl=0 works just fine with a simple workaround (see below). 
Running  `java -jar owlcms_4.2.9_withWorkaround.jar` is the expected behavior -- an embedded jetty starts without requiring an npm run and a web browser opens.

- Proposed workaround
Remove the test for >=2 in `com.vaadin.flow.server.DeploymentConfigurationFactory.getPossibleJarResource(List<URL>)` .
The logic would be correct if the application is compiled as a jar, and deployed with all the other dependencies in their own jars, in a long classpath.
This is not the case with a uber-jar created with maven-shade-plugin, for example.  All the classes and files are in a single file.
The following code works *for a uber-jar* -- I have not analyzed the consequences of checking the == 1 for the "long classpath with many jars" cases.  Therefore I am not proposing this as a pull request without further thought.
```
    private static String getPossibleJarResource(List<URL> resources) throws IOException {
            for (URL resource : resources) {
                // As we now know that we are running from a jar we can accept a
                // build info with a single jar in the path
                if (countInstances(resource.getPath(), "jar!/") == 1) {
                    return FrontendUtils.streamToString(resource.openStream());
                }
            }
        return null;
    }
```
- Notes: there appears to be several cases
  - development with individual classes and folders
  - long classpath with application packaged as a jar and other jars for dependencies
  - uber-jar/fat-jar with everything in the same jar.

- Versions:
    - Vaadin / Flow version: 14.0.13
    - Java version: AdoptOpenJDK 8 and 11
    - OS version: n/a
    - Browser version (if applicable): n/a

**Comment by caalador** [comment]
This is a duplicate of #7060 for a flattened jar and a fix (#7072) has been added and published in Flow 2.1.1 that should be out with 14.1.2 today. 
This will now try to determine a valid build info, but accept any found at the end.

**Comment by caalador** [comment]
Closing issue as 7060 was verified.
